site/src/data/releases/v0.4.json

```diff
@@ -0,0 +1,301 @@
+            },
+            {
+              "title": "Guided choice support",
+              "description": "Restricts model outputs to predefined choices for GCP Vertex AI/Gemini models, ensuring responses conform to expected values."
+            },
+            {
+              "title": "Guided JSON support",
+              "description": "Ensures model outputs are valid JSON conforming to specified schemas for GCP Vertex AI/Gemini models, with OpenAI-compatible API translation."
+            }
+          ]
+        },
+        {
+          "title": "Cross-Namespace Resource References",
+          "items": [
+            {
+              "title": "Cross-namespace <code>AIServiceBackend</code> references",
+              "description": "<code>AIGatewayRoute</code> can now reference <code>AIServiceBackend</code> resources in different namespaces, enabling multi-tenant and organizational separation patterns."
+            },
+            {
+              "title": "ReferenceGrant validation",
+              "description": "Comprehensive ReferenceGrant integration following Gateway API patterns, with automatic validation and clear error messages when grants are missing."
+            },
+            {
+              "title": "Event-driven reconciliation",
+              "description": "ReferenceGrant changes automatically trigger reconciliation of affected routes, ensuring security policies are always up-to-date."
+            }
+          ]
+        },
+        {
+          "title": "Enhanced Upstream Authentication",
+          "items": [
+            {
+              "title": "AWS SDK default credential chain",
+              "description": "Support for AWS SDK's default credential chain including IRSA (IAM Roles for Service Accounts), EKS Pod Identity, EC2 Instance Profiles, and environment variables, eliminating need for static credentials."
+            },
+            {
+              "title": "Azure API key authentication",
+              "description": "Native Azure OpenAI API key authentication using the <code>api-key</code> header, matching Azure SDK conventions and console practices."
+            },
+            {
+              "title": "Azure CLI auto-configuration",
+              "description": "Standalone CLI auto-configuration from <code>AZURE_OPENAI_API_KEY</code> and <code>AZURE_OPENAI_ENDPOINT</code> environment variables, compatible with Elastic and other Azure-based deployments."
+            }
+          ]
+        },
+        {
+          "title": "Traffic Management and Configuration",
+          "items": [
+            {
+              "title": "Route-level header mutations",
+              "description": "New <code>headerMutation</code> field in <code>AIGatewayRouteRuleBackendRef</code> enables per-route header manipulation with smart merge logic when combined with backend-level mutations. Essential for GCP provisioned throughput and other advanced routing scenarios."
+            },
```

Comment:
not only the route level but also wehave AIServiceBackend level (https://github.com/envoyproxy/ai-gateway/commit/4bfaf9135c93d6de9599846a1b676e62bb67f4d1) one, and they are both new in v0.4.